projects/packages/classic-theme-helper/src/content-options/customizer.php

```diff
@@ -0,0 +1,521 @@
+					)
+				);
+
+				$wp_customize->add_control(
+					'jetpack_content_featured_images_fallback',
+					array(
+						'section'         => 'jetpack_content_options',
+						'label'           => esc_html__( 'Automatically use first image in post', 'jetpack-classic-theme-helper' ),
+						'type'            => 'checkbox',
+						'active_callback' => 'jetpack_post_thumbnail_supports',
+					)
+				);
+			}
+		}
+	}
+	add_action( 'customize_register', 'jetpack_content_options_customize_register' );
+
+}
+
+if ( ! function_exists( 'jetpack_post_thumbnail_supports' ) ) {
+
+	/**
+	 * Return whether the theme supports Post Thumbnails.
+	 */
+	function jetpack_post_thumbnail_supports() {
+		return ( current_theme_supports( 'post-thumbnails' ) );
+	}
+
+}
+
+if ( ! function_exists( 'jetpack_content_options_sanitize_checkbox' ) ) {
+
+	/**
+	 * Sanitize the checkbox.
+	 *
+	 * @param int $input The unsanitized value from the setting.
+	 * @return int|string
+	 */
+	function jetpack_content_options_sanitize_checkbox( $input ) {
+		return ( 1 === (int) $input ) ? 1 : '';
+	}
+
+}
+
+if ( ! function_exists( 'jetpack_content_options_sanitize_blog_display' ) ) {
+
+	/**
+	 * Sanitize the Display value.
+	 *
+	 * @param string $display The unsanitized value from the setting.
+	 * @return string
+	 */
+	function jetpack_content_options_sanitize_blog_display( $display ) {
+		if ( ! in_array( $display, array( 'content', 'excerpt', 'mixed' ), true ) ) {
+			$display = 'content';
+		}
+		return $display;
+	}
+
+}
+
+if ( ! function_exists( 'jetpack_content_options_customize_preview_js' ) ) {
+
+	/**
+	 * Binds JS handlers to make Theme Customizer preview reload changes asynchronously.
+	 */
+	function jetpack_content_options_customize_preview_js() {
+		$options      = get_theme_support( 'jetpack-content-options' );
+		$blog_display = ( ! empty( $options[0]['blog-display'] ) ) ? $options[0]['blog-display'] : null;
+		$blog_display = preg_grep( '/^(content|excerpt)$/', (array) $blog_display );
+		sort( $blog_display );
+		$blog_display = implode( ', ', $blog_display );
+		$blog_display = ( 'content, excerpt' === $blog_display ) ? 'mixed' : $blog_display;
+		$masonry      = ( ! empty( $options[0]['masonry'] ) ) ? $options[0]['masonry'] : null;
+		$post_details = ( ! empty( $options[0]['post-details'] ) ) ? $options[0]['post-details'] : null;
+		$date         = ( ! empty( $post_details['date'] ) ) ? $post_details['date'] : null;
+		$categories   = ( ! empty( $post_details['categories'] ) ) ? $post_details['categories'] : null;
+		$tags         = ( ! empty( $post_details['tags'] ) ) ? $post_details['tags'] : null;
+		$author       = ( ! empty( $post_details['author'] ) ) ? $post_details['author'] : null;
+		$comment      = ( ! empty( $post_details['comment'] ) ) ? $post_details['comment'] : null;
+
+		wp_enqueue_script( 'jetpack-content-options-customizer', plugins_url( 'customizer.js', __FILE__ ), array( 'jquery', 'customize-preview' ), '1.0', true );
```

Comment:
It would be worth looking into whether the Assets package can be used here for any scripts, where possible. Particularly as Webpack is compiling a `min.js` version of `customizer.js` as well. I'll look into that when requiring the feature via the package so it can be tested.
Also possibly relevant for `post-details.php` as well.